route.py

```diff
@@ -199,11 +202,18 @@ def try_update_default(self, routes):
 
         current = self.get_default()
         try:
-            if current["interface"] != default["interface"] or \
-                    current["gateway"] != default["gateway"]:
+            if current != default:
                 self.update_default(default)
-        except:
-            self.update_default(default)
+        except Exception as e:
+            _logger.debug(e)
+            try:
+                self.update_default(default)
+            except:
```

Comment:
We should always except some specific Errors otherwise it would be much harder to debug without log anything and pass all exceptions.